sys/cpp11-compat/include/riot/condition_variable.hpp

```diff
@@ -176,12 +177,13 @@ cv_status condition_variable::wait_for(unique_lock<mutex>& lock,
   timeout.seconds = s.count();
   timeout.microseconds
     = (duration_cast<microseconds>(timeout_duration - s)).count();
-  xtimer_now_timex(&before);
-  xtimer_t timer;
-  xtimer_set_wakeup(&timer, timex_uint64(timeout), thread_getpid());
+  before = timex_from_uint64(ztimer64_now(ZTIMER64_USEC));
```

Comment:
Why is duration cast to timex_t (l176 to 179) ? (not to 64bit microseconds directly)